apollo-router/src/plugins/telemetry/metrics/span_metrics_exporter.rs

```diff
@@ -39,55 +37,104 @@ impl Default for Exporter {
     }
 }
 
-#[async_trait]
-impl SpanExporter for Exporter {
-    /// Export spans metrics to real metrics
-    fn export(&mut self, batch: Vec<SpanData>) -> BoxFuture<'static, ExportResult> {
-        for span in batch
-            .into_iter()
-            .filter(|s| self.span_names.contains(&s.name.as_ref()))
-        {
-            let busy = span
-                .attributes
-                .get(&BUSY_NS_ATTRIBUTE_NAME)
-                .and_then(|attr| match attr {
-                    Value::I64(v) => Some(*v),
-                    _ => None,
-                })
-                .unwrap_or_default();
-            let idle = span
-                .attributes
-                .get(&IDLE_NS_ATTRIBUTE_NAME)
-                .and_then(|attr| match attr {
-                    Value::I64(v) => Some(*v),
-                    _ => None,
-                })
-                .unwrap_or_default();
-            let duration = span
-                .end_time
-                .duration_since(span.start_time)
-                .unwrap_or_default()
-                .as_secs_f64();
+impl<S> Layer<S> for SpanMetricsLayer
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    fn on_new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, ctx: Context<'_, S>) {
+        let span = ctx.span(id).expect("Span not found, this is a bug");
```

Comment:
Not comfortable with an expect here, are we 100% sure this wont ever ever happen ever for anyone ?

Reply:
for some context, this is directly taken from `OpenTelemetryLayer` so thereis no additional risk we are taking right now, since that layer is always active https://github.com/tokio-rs/tracing-opentelemetry/blob/4fa752748116f37af82b59e9b74cb042afa035c1/src/layer.rs#L672